task-sdk/docs/index.rst

```diff
@@ -0,0 +1,161 @@
+- ``@task_group``: group related tasks into logical units.
+- ``@setup`` and ``@teardown``: define setup and teardown tasks for DAGs and TaskGroups.
+
+.. literalinclude:: ../../airflow-core/src/airflow/example_dags/example_task_group_decorator.py
+   :language: python
+   :start-after: [START howto_task_group_decorator]
+   :end-before: [END howto_task_group_decorator]
+   :caption: Group tasks using the ``@task_group`` decorator.
+
+.. literalinclude:: ../../airflow-core/src/airflow/example_dags/example_setup_teardown_taskflow.py
+   :language: python
+   :caption: Define setup and teardown tasks with ``@setup`` and ``@teardown``.
+
+Tasks and Operators
+~~~~~~~~~~~~~~~~~~~
+Wrap Python callables with ``@task`` to create tasks, leverage dynamic task mapping with
+``.expand()``, and pass data via ``XComArg``. You can also create traditional Operators
+(e.g., sensors) via classes imported from the SDK:
+
+  - **BaseOperator**, **Sensor**, **OperatorLink**, **Notifier**, "XComArg", etc.
+    (see the **api reference** section for details)
+
+.. literalinclude:: ../../airflow-core/src/airflow/example_dags/example_dynamic_task_mapping.py
+   :language: python
+   :start-after: [START example_dynamic_task_mapping]
+   :end-before: [END example_dynamic_task_mapping]
+   :caption: Dynamic task mapping with ``expand()``
+
+.. literalinclude:: ../../airflow-core/src/airflow/example_dags/example_xcomargs.py
+   :language: python
+   :caption: Using ``XComArg`` to chain tasks based on return values.
+
+Assets
+~~~~~~
+Model data as assets and emit them to downstream tasks with the SDK's asset library under
+``airflow.sdk.definitions.asset``. You can use:
+
+  - ``@asset``, ``AssetAlias``, etc. (see the **api reference** section below)
+
+.. literalinclude:: ../../airflow-core/src/airflow/example_dags/example_assets.py
+   :language: python
+   :start-after: [START asset_def]
+   :end-before: [END asset_def]
+   :caption: Defining an ``Asset``
+
+.. literalinclude:: ../../airflow-core/src/airflow/example_dags/example_asset_alias.py
+   :language: python
+   :caption: Defining asset aliases with ``AssetAlias``.
+
+Execution Time Components
+~~~~~~~~~~~~~~~~~~~~~~~~~
+At runtime, tasks run in an isolated subprocess managed by the SDK:
+
+  - **Supervisor** coordinates the worker's lifecycle.
+  - **TaskRunner** actually executes the user's task code.
+  - **Context** objects provide runtime metadata (e.g., connections, variables).
+    (see the **Execution Time** section below for details)
+
+
+Everything Else
+---------------
+In addition to "DAG" and "task"-level decorators, the Task SDK provides:
+
+  * **Bases** (under ``airflow.sdk.bases``):
+    - ``BaseOperator``
+    - ``Sensor``
+    - ``OperatorLink``
+    - ``Notifier``
+    - ``XComArg``
+
+  * **Decorators**:
+    - ``@task``, ``@task_group``, ``@setup``, ``@teardown`` (exported at top level).
+    - These live in ``airflow.sdk.decorator`` but are re-exported in ``airflow.sdk``.
```

Comment:
Don't even _tell_ users about `airflow.sdk.decorator` module.

Reply:
Removed it